src/inmanta/util/dict_path.py

```diff
@@ -33,6 +33,34 @@
 TWCP = TypeVar("TWCP", bound="WildComposedPath")
 
 
+@runtime_checkable
+class Mapping(Protocol):
```

Comment:
The reason we did this is that the meta class of resource and abc.Mapping don't get along